test/common_device_type.py

```diff
@@ -364,6 +391,32 @@ def multi_fn(slf, devices, *args, **kwargs):
         return multi_fn
 
 
+# Specifies per-dtype precision overrides.
+# Ex.
+#
+# @overridePrecision(torch.half : 1e-2, torch.float : 1e-4)
+# @dtypes(torch.half, torch.float, torch.double)
+# def test_X(self, device, dtype):
+#   ...
+#
+# When the test is instantiated its class's precision will be set to the
+# corresponding override, if it exists.
+# self.precision can be accessed directly, and it also controls the behavior of
```

Comment:
nit: I'd say a little more in that there's only a single precision, so even if you compare other dtypes (you can make a note that this is non-standard and not recommended), it will test with _the_ precision.